include/swift/Reflection/ReflectionContext.h

```diff
@@ -378,30 +378,29 @@ class ReflectionContext
     return readPECOFFSections(ImageStart);
   }
 
-  template <typename T> bool readELFSections(RemoteAddress ImageStart) {
+  template <typename T> bool readELFSections(RemoteAddress ImageStart, void *buffer) {
     auto Buf =
         this->getReader().readBytes(ImageStart, sizeof(typename T::Header));
 
     auto Hdr = reinterpret_cast<const typename T::Header *>(Buf.get());
     assert(Hdr->getFileClass() == T::ELFClass && "invalid ELF file class");
 
     // From the header, grab informations about the section header table.
-    auto SectionHdrAddress = ImageStart.getAddressData() + Hdr->e_shoff;
+    auto SectionHdrAddress = (unsigned long)buffer + Hdr->e_shoff;
     auto SectionHdrNumEntries = Hdr->e_shnum;
     auto SectionEntrySize = Hdr->e_shentsize;
 
     // Collect all the section headers, we need them to look up the
     // reflection sections (by name) and the string table.
     std::vector<const typename T::Section *> SecHdrVec;
     for (unsigned I = 0; I < SectionHdrNumEntries; ++I) {
-      auto SecBuf = this->getReader().readBytes(
-          RemoteAddress(SectionHdrAddress + (I * SectionEntrySize)),
-          SectionEntrySize);
+      void *SecBuf = malloc(SectionEntrySize);
```

Comment:
Since this is C++ we might want to use something with automatic memory management here (`std::vector<char>` ?)